src/rules/typedefWhitespaceRule.ts

```diff
@@ -14,129 +14,102 @@
-/// <reference path="../../lib/tslint.d.ts" />
+///<reference path="../../lib/tslint.d.ts" />
 
 export class Rule extends Lint.Rules.AbstractRule {
     public static FAILURE_STRING = "missing type declaration";
 
-    public apply(syntaxTree: TypeScript.SyntaxTree): Lint.RuleFailure[] {
-        return this.applyWithWalker(<Lint.RuleWalker>(new TypedefWhitespaceWalker(syntaxTree, this.getOptions())));
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        return this.applyWithWalker(new TypedefWhitespaceWalker(sourceFile, this.getOptions()));
     }
 }
 
 class TypedefWhitespaceWalker extends Lint.RuleWalker {
-
-    public visitCallSignature(node: TypeScript.CallSignatureSyntax): void {
-        this.checkSpace("call-signature", node, node.typeAnnotation);
-
-        super.visitCallSignature(node);
+    public visitFunctionDeclaration(node: ts.FunctionDeclaration) {
+        this.checkSpace("call-signature", node, node.type, node.parameters.end + 1);
+        super.visitFunctionDeclaration(node);
     }
 
-    public visitCatchClause(node: TypeScript.CatchClauseSyntax): void {
-        this.checkSpace("catch-clause", node, node.typeAnnotation);
+    public visitFunctionExpression(node: ts.FunctionExpression) {
+        this.checkSpace("call-signature", node, node.type, node.parameters.end + 1);
+        super.visitFunctionExpression(node);
+    }
 
-        super.visitCatchClause(node);
+    public visitGetAccessor(node: ts.AccessorDeclaration) {
+        this.checkSpace("call-signature", node, node.type, node.parameters.end + 1);
+        super.visitGetAccessor(node);
     }
 
-    public visitIndexSignature(node: TypeScript.IndexSignatureSyntax): void {
-        this.checkSpace("index-signature", node, node.typeAnnotation);
+    public visitIndexSignatureDeclaration(node: ts.IndexSignatureDeclaration) {
+        var indexParameter = node.parameters[0];
 
-        super.visitIndexSignature(node);
+        if (indexParameter != null) {
+            this.checkSpace("index-signature", indexParameter, indexParameter.type, indexParameter.name.getEnd());
+        }
+        super.visitIndexSignatureDeclaration(node);
     }
 
-    public visitParameter(node: TypeScript.ParameterSyntax): void {
-        this.checkSpace("parameter", node, node.typeAnnotation);
-
-        super.visitParameter(node);
+    public visitMethodDeclaration(node: ts.MethodDeclaration) {
+        this.checkSpace("call-signature", node, node.type, node.parameters.end + 1);
+        super.visitMethodDeclaration(node);
     }
 
-    public visitPropertySignature(node: TypeScript.PropertySignatureSyntax): void {
-        this.checkSpace("property-signature", node, node.typeAnnotation);
+    public visitParameterDeclaration(node: ts.ParameterDeclaration) {
+        // a parameter's "type" could be a string literal, for example `fn(option: "someOption", anotherOption: number)`
+        if (node.type != null && node.type.kind !== ts.SyntaxKind.StringLiteral) {
```

Comment:
You should probably check for equality rather than checking it isn't the other one here, it's cleaner. Also, your comment above checkSpace below implies it can handle string literals here -- why are you doing this, if that's the case?